### PR TITLE
Add environment vars cli flag to stack init

### LINKED ISSUE #302: firefly docker compose stack through a corporate proxy having problems
The current docker compose stack generated by `ff init` inherits the docker proxy settings directly from the host. This causes intercommunication between the containers to fail since the proxy is unable to resolve the DNS names of the compose containers properly.  While it is possible to configure the variables at the host/docker service level, this causes other docker containers outside of the compose to have issues. 

Would it be possible to have generic flag in the firefly CLI to allow the setting of generic environment variables across all containers within the docker compose? One potential use case would be to unset `HTTP_PROXY` variable. I'll be happy to work on this PR.

### PR DESCRIPTION
**Pull request by rodion-lim-partior**
This PR adds a `--environment-vars` flag to the **init** command to allow docker compose generation to include generic environment variables across all containers. When docker is started against a corporate proxy, containers are unable to resolve via the internal DNS names due to the proxy. 

In this instance, users need to unset the proxy by passing in environment variables `HTTP_PROXY=,http_proxy=` to the containers. Depending on needs, other proxy specific environment variables can be set as well.

Resolves #302

### CODE REVIEW

File: pkg/types/stack.go [hunk 1]
```diff
@@ -106,3 +107,14 @@ func (s *Stack) IsOldFileStructure() (bool, error) {
 		return false, nil
 	}
 }
+
+func (s *Stack) ConcatenateEnvironmentVars(input map[string]interface{}) map[string]interface{} {
```
Comment (EnriqueL8) [inline pkg/types/stack.go:111]: nit: rename to `ConcatenateWithProvidedEnvironmentVars`

File: cmd/init.go [hunk 2]
```diff
@@ -271,5 +271,6 @@ func init() {
 	initCmd.PersistentFlags().StringArrayVar(&initOptions.OrgNames, "org-name", []string{}, "Organization name")
 	initCmd.PersistentFlags().StringArrayVar(&initOptions.NodeNames, "node-name", []string{}, "Node name")
 	initCmd.PersistentFlags().BoolVar(&initOptions.RemoteNodeDeploy, "remote-node-deploy", false, "Enable or disable deployment of FireFly contracts on remote nodes")
+	initCmd.PersistentFlags().StringToStringVar(&initOptions.EnvironmentVars, "environment-vars", map[string]string{}, "Common environment variables to set on all containers in FireFly stack")
```
Comment (EnriqueL8) [inline cmd/init.go:274]: I wonder if we should keep it as a `map[string]interface{}` ?

**Review by EnriqueL8 — COMMENTED** [review]
Thanks for the contribution @rodion-lim-partior, I like were this is going!

I'm wondering if you have tested the behaviour when using the `docker-composer.override.yml` and overriding the environment variables for a specific container and how that would impact the environment variables set here. I also only see the value of this feature for things such as making all the containers use a PROXY, or networking setup? Most of the time users will want to customise specific environment variables for each service and the `docker-compose.override.yml` is probably the best solution.

**Comment by rodion-lim-partior** [comment]
@EnriqueL8, I tested locally with the docker compose overrides. docker compose have precedence over CLI options overrides, which is what we expect. Also, the docker compose overrides do not wipe CLI options, it merges the difference in.

This are the EnvironmentVars that we pass into the CLI options
```
"HTTP_PROXY=,http_proxy="
```

This is the docker compose overrides used:
```
version: "2.1"
services:
  geth_1:
    environment:
      no_proxy: ""
      HTTP_PROXY: "abc"
```
In geth_0 container:
/ # env | grep -i proxy
no_proxy=127.0.0.0/8
HTTP_PROXY=
http_proxy=

In geth_1 container:
/ # env | grep -i proxy
no_proxy=
HTTP_PROXY=abc
http_proxy=

Also, you are right that in our use case, the value of this feature is for things such as making all the containers point to the same proxy variables.

File: pkg/types/stack.go (hunk 1 above)
Comment (rodion-lim-partior) [reply pkg/types/stack.go]: renamed

File: cmd/init.go (hunk 2 above)
Comment (rodion-lim-partior) [reply cmd/init.go:274]: I think we should keep this as a string. Environment variables won't be nested, they are always in the format var_name=value. Also, they will always be in string format as well, so we should keep the type safety.

**Review by EnriqueL8 — APPROVED** [review]
I'm happy with this one going in

**Pull request merged**